### PR TITLE
cta attributes from link & button

### PR DESCRIPTION
**Pull request by forresto**
`<a href="https://link.com/" data-role="cta">Call to action!</a>` makes a `cta` block.

### CODE REVIEW

**Comment by forresto** [comment]
@bergie how is this for a start?

**Comment by bergie** [comment]
Here is example of a CTA button: https://raw.githubusercontent.com/the-domains/the-grid/master/_posts/2014-10-05-join-the-evolution.md

**Comment by bergie** [comment]
We need CTA blocks to provide similar data structure as https://github.com/the-grid/apidocs/blob/master/schemata/sitepurpose.yaml

**Comment by forresto** [comment]
@bergie Do we need to support those fields now or later?

**Comment by bergie** [comment]
@forresto on html-flatten side we should support all of them. In Ed maybe not.

**Comment by forresto** [comment]
So same deal as in `button`? https://github.com/the-grid/html-flatten/blob/7eb20df744d9812b66c440619d9b6abb51d6e50f/index.coffee#L375-L379 ... or whitelist?

**Comment by forresto** [comment]
The widget needs a whitelist to render the fields... so I'll do that here.

File: index.coffee [hunk 1]
```diff
@@ -364,6 +364,10 @@ module.exports = class Flatten
         return results unless normalizedChildren.length
         if tag.attribs?.href
           normalizedChildren[0].html = @tagToHtml tag, id
+        if tag.attribs?['data-role'] is 'cta' and tag.attribs.href
+          normalizedChildren[0].type = 'cta'
+          normalizedChildren[0].url = tag.attribs.href
+          normalizedChildren[0].label = @tagToText(tag)
```
Comment (bergie) [inline index.coffee:372]: We need support for linking to a Reserve CTA from attributes, like in https://github.com/the-grid/apidocs/blob/master/schemata/sitepurpose.yaml#L22

**Comment by forresto** [comment]
@bergie there is a conflict with cta type and block type... 
- `data-role="cta"` → `block.type: "cta"` 
- `data-type="purchase"` → `block.type: "purchase"`

**Comment by bergie** [comment]
@forresto ah, yes. Stupid of me. We used `verb` for the latter. So:
- `block.type: 'cta'`
- `block.verb: 'purchase'`

Should probably be based on `data-verb` not `data-type` as well.

**Comment by forresto** [comment]
I'll fix/spec button too then.

**Comment by forresto** [comment]
@bergie 👀

**Comment by bergie** [comment]
@forresto thanks! 🚢

**Pull request merged**